File: perl/operators/operators_increment_and_decrement.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my $a = 6; my $b = 3;

print("\$a = ${a}, \$b = ${b}\n");
# preincrement
print("++\$a : " . (++$a) . "(" . $a . ")\n");
# predecrement
print("--\$b : " . (--$b) . "(" . $b . ")\n");

print("\$a = ${a}, \$b = ${b}\n");
# postincrement
print("\$a++ : " . ($a++) . "(" . $a . ")\n");
# postdecrement
print("\$b-- : " . ($b--) . "(" . $b . ")\n");

File: c#/operators/OperatorsIncrementAndDecrement.cs
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class OperatorsIncrementAndDecrement {
  public static void Main(string[] args) {
    int a = 6, b = 3;

    System.Console.WriteLine(string.Format("a = {0}, b = {1}", a, b));
    // preincrement
    System.Console.WriteLine(string.Format("++a : {0}({1})", ++a, a));
    // predecrement
    System.Console.WriteLine(string.Format("--b : {0}({1})", --b, b));

    System.Console.WriteLine(string.Format("a = {0}, b = {1}", a, b));
    // postincrement
    System.Console.WriteLine(string.Format("a++ : {0}({1})", a++, a));
    // postdecrement
    System.Console.WriteLine(string.Format("b-- : {0}({1})", b--, b));
  }
}
